### PR TITLE
Ability to display the search phrase in the title

### PR DESCRIPTION
**Pull request by bogdasar1985**
https://github.com/newsboat/newsboat/issues/2043

### CODE REVIEW

**Comment by bogdasar1985** [comment]
That is also should be documented.

File: src/searchresultslistformaction.cpp [hunk 1]
```diff
@@ -89,4 +89,33 @@ std::string SearchResultsListFormAction::title()
...
+
+	auto feedtitle = s;
+	utils::remove_soft_hyphens(feedtitle);
+	fmt.register_fmt('T', feedtitle);
+
+	fmt.register_fmt('U', utils::censor_url(url));
+
+	fmt.register_fmt('F', apply_filter ? matcher.get_expression() : "");
+
+	fmt.register_fmt('s', search_phrase);
```
Comment (Minoru) [inline src/searchresultslistformaction.cpp:114]: Why copy the entire method? I think one could just call the base method and add to its result:

```c++
FmtStrFormatter fmt = ItemListFormAction::setup_head_formatter(…);
fmt.register_fmt('s', search_phrase);
return fmt;
```

**Review by Minoru — CHANGES_REQUESTED** [review]
Sorry for the slow response! I think the code could be made more maintainable by avoiding copying, see my comment below.

I also agree that this should be documented and added to the default `searchresult-title-format`. Please do that :)

**Review by Minoru — CHANGES_REQUESTED** [review]
The code looks good, now all that's left is to document this and change the default.

File: doc/newsboat.asciidoc [hunk 2]
```diff
@@ -1171,6 +1171,14 @@ Identifier:Meaning
 [[articleview-title-format-t]]<<articleview-title-format-t,+t+>>:Number of total articles
 |======================================================================
 
+.Search Result Title Format Identifiers
```
Comment (Minoru) [inline doc/newsboat.asciidoc:1174]: I'd move this table up, right after the one for the article list, and added a note that all of the article list identifiers are supported here too.

File: doc/configcommands.dsv [hunk 3]
```diff
@@ -129,7 +129,7 @@ run-on-startup||<list of operations>||n/a||Specifies one or more <<_newsboat_ope
 save-path||<path-to-directory>||~/||The default path where articles shall be saved to. If an invalid path is specified, the current directory is used.||save-path "~/Saved Articles"
 scrolloff||<number>||0||Keep the configured number of lines above and below the selected item in lists. Configure a high number to keep the selected item in the center of the screen.||scrolloff 5
 search-highlight-colors||<fgcolor> <bgcolor> [<attribute> ...]||black yellow bold||This configuration command specifies the highlighting colors when searching for text from the article view.||search-highlight-colors white black bold
-searchresult-title-format||<format>||"%N %V - Search results (%u unread, %t total)%?F? matching filter '%F'&?" (localized)||Format of the title in search result. See "Format Strings" section of Newsboat manual for details on available formats.||searchresult-title-format "Search result"
+searchresult-title-format||<format>||"%N %V - Search results:\"%s\" (%u unread, %t total)%?F? matching filter '%F'&?" (localized)||Format of the title in search result. See "Format Strings" section of Newsboat manual for details on available formats.||searchresult-title-format "Search result"
```
Comment (Minoru) [inline doc/configcommands.dsv:132]:
```suggestion
searchresult-title-format||<format>||"%N %V - Search results for '%s' (%u unread, %t total)%?F? matching filter '%F'&?" (localized)||Format of the title in search result. See "Format Strings" section of Newsboat manual for details on available formats.||searchresult-title-format "Search result"
```

I think this reads better, and the quotes are consistent with what we use for the filter.

File: doc/newsboat.asciidoc [hunk 4]
```diff
@@ -1104,7 +1104,7 @@ title text may vary unless you customize it.
 Dialog:Configuration Variable:Default Value
 Feed List:<<feedlist-title-format,+feedlist-title-format+>>:%N %V - %?F?Feeds&Your feeds? (%u unread, %t total)%?F? matching filter `%F'&?%?T? - tag `%T'&?
 Article List:<<articlelist-title-format,+articlelist-title-format+>>:%N %V - Articles in feed '%T' (%u unread, %t total)%?F? matching filter `%F'&? - %U
-Search Result:<<searchresult-title-format,+searchresult-title-format+>>:%N %V - Search results (%u unread, %t total)%?F? matching filter `%F'&?
+Search Result:<<searchresult-title-format,+searchresult-title-format+>>:%N %V - Search results:\"%s\" (%u unread, %t total)%?F? matching filter `%F'&?
```
Comment (Minoru) [inline doc/newsboat.asciidoc:1107]:
```suggestion
Search Result:<<searchresult-title-format,+searchresult-title-format+>>:%N %V - Search results for '%s' (%u unread, %t total)%?F? matching filter `%F'&?
```

File: src/configcontainer.cpp [hunk 5]
```diff
@@ -312,7 +312,7 @@ ConfigContainer::ConfigContainer()
 			ConfigDataType::STR)},
 	{
 		"searchresult-title-format",
-		ConfigData(_("%N %V - Search results (%u unread, %t total)%?F? matching filter '%F'&?"),
+		ConfigData(_("%N %V - Search results:\"%s\" (%u unread, %t total)%?F? matching filter '%F'&?"),
```
Comment (Minoru) [inline src/configcontainer.cpp:315]:
```suggestion
		ConfigData(_("%N %V - Search results for '%s' (%u unread, %t total)%?F? matching filter '%F'&?"),
```

**Comment by Minoru** [comment]
Almost there! :)

**Review by Minoru — APPROVED** [review]
You forgot to add a note that `searchresult-title-format` supports the same identifiers as `articlelist-title-format`. When marking a review comment as resolved, please make sure it's *really* resolved ;)

Anyway, this looks excellent and I'm merging. Thanks for the work!

**Comment by bogdasar1985** [comment]
@Minoru Great! Please, check IRC channel. I suggested something there. I will open an issue if there is no answer.

**Pull request merged**